### PR TITLE
Add Ulog sync bytes to ulog stream

### PR DESCRIPTION
**Pull request by devbharat**
To go along with https://github.com/PX4/pyulog/pull/41

**Describe problem solved by the proposed pull request**
Send ulog sync bytes at 2Hz to recover from ulog data corruptions. If we want, we could have the rate as a param.

### CODE REVIEW

File: src/modules/logger/messages.h [hunk 1]
```diff
@@ -44,6 +44,7 @@ enum class ULogMessageType : uint8_t {
 	SYNC = 'S',
 	DROPOUT = 'O',
 	LOGGING = 'L',
+	LOGGING_OBC = 'C',
```
[the remark this answers is not in the document]
Comment (devbharat) [reply src/modules/logger/messages.h]: It is, but keeping it here 'reserves' it as it is used for that purpose in pyulog (so nobody uses that byte for some other purpose). Would have been nicer if this was somehow autogenerated and used in both PX4 and pyulog and did not need explicit syncing.

File: src/modules/logger/logger.cpp [hunk 2]
```diff
@@ -1190,6 +1190,29 @@ void Logger::run()
 				}
 			}
 
+			// Add sync magic
+			if ((loop_time - _last_sync_time) > 5e5) {
+				ulog_message_sync_s sync_message;
```
[the remark this answers is not in the document]
Comment (devbharat) [reply src/modules/logger/logger.cpp]: true

File: src/modules/logger/logger.cpp [hunk 3]
```diff
@@ -1190,6 +1190,29 @@ void Logger::run()
 				}
 			}
 
+			// Add sync magic
+			if ((loop_time - _last_sync_time) > 5e5) {
```
[the remark this answers is not in the document]
Comment (devbharat) [reply src/modules/logger/logger.cpp]: this broke it

File: src/modules/logger/messages.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (bkueng) [reply src/modules/logger/messages.h]: Ok, in that case there's a missing `ulog_message_logging_tagged_s` struct definition.

File: src/modules/logger/logger.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (bkueng) [reply src/modules/logger/logger.cpp]: How?

File: src/modules/logger/messages.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (devbharat) [reply src/modules/logger/messages.h]: Done

File: src/modules/logger/logger.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (devbharat) [reply src/modules/logger/logger.cpp]:
```
../../src/modules/logger/logger.cpp:1194:38: error: unable to find numeric literal operator 'operator""_ms'

    if (loop_time - _last_sync_time > 500_ms) {

                                      ^~~~~~

compilation terminated due to -Wfatal-errors.

```

[the remark this answers is not in the document]
Comment (bkueng) [reply src/modules/logger/logger.cpp]: Add `using namespace time_literals;` at the top of the file.

**Pull request merged**